Repository/RepoLogic.cs

```diff
@@ -0,0 +1,145 @@
+        }
+
+        /// <summary>
+        /// Returns a list of all User objects in the database. If there are no users,
+        /// the list will be empty.
+        /// </summary>
+        /// <returns></returns>
+        public async Task<List<User>> GetUsers()
+        {
+            return await _dbContext.Users.ToListAsync();
+        }
+
+        /// <summary>
+        /// Returns a list of all FollowingMovie objects from the database that match the username
+        /// specified in the argument. Returns null if the user doesn't exist.
+        /// </summary>
+        /// <param name="username"></param>
+        /// <returns></returns>
+        public async Task<List<FollowingMovie>> GetFollowingMovies(string username)
+        {
+            var userExists = UserExists(username);
+            if(!userExists)
+            {
+                Console.WriteLine("RepoLogic.GetFollowingMovies() was called for a user that doesn't exist.");
+                return null;
+            }
+            return await _dbContext.FollowingMovies.Where(f => f.Username == username).ToListAsync();
+        }
+
+        /// <summary>
+        /// Adds the FollowingMovie specified in the argument to the database.
+        /// Returns true iff successful.
+        /// Returns false if the username or movieid referenced in the FollowingMovie object
+        /// do not already exist in their respective database tables.
+        /// </summary>
+        /// <param name="followingMovie"></param>
+        /// <returns></returns>
+        public async Task<bool> FollowMovie(FollowingMovie followingMovie)
+        {
+            var userExists = UserExists(followingMovie.Username);
+            if(!userExists)
+            {
+                Console.WriteLine("RepoLogic.FollowMovie() was called for a user that doesn't exist.");
+                return false;
+            }
+// MS API CALL: Make sure movie exists
+
+            // Ensure the User is not already Following this Movie
+            if((await _dbContext.FollowingMovies.Where(fm => 
+                    fm.Username == followingMovie.Username 
+                    && fm.MovieId == followingMovie.MovieId
+                ).FirstOrDefaultAsync<FollowingMovie>()) != null)
+            {
+                Console.WriteLine("RepoLogic.FollowMovie() was called for a movie that the user is " +
+                    "already following.");
+                return false;
+            }
+
+            await _dbContext.FollowingMovies.AddAsync(followingMovie);
+
+            await _dbContext.SaveChangesAsync();
+            return true;
+        }
+
+        /// <summary>
+        /// Returns true iff the username, specified in the argument, exists in the database's Users table.
+        /// </summary>
+        /// <param name="username"></param>
+        /// <returns></returns>
+        private bool UserExists(string username)
```

Comment:
Is this method used for logging in? If so, you should be using a combo username and password to find the user. Of course, separate methods for unique username and unique password would be more reusable.